notebooks/Tutorial9-Multitable.py

```diff
@@ -0,0 +1,521 @@
+axes[1].set_title("Job Distribution")
+plt.tight_layout()
+plt.show()
+# -
+
+# ### Visit
+
+visit_combined = pd.concat([visit, visit_avatar]).reset_index(drop=True)
+visit_combined["type"] = np.repeat(["original", "avatar"], 300)
+map_color = {"original": "dimgrey", "avatar": "#3BD6B0"}
+fig, axes = plt.subplots(1, 3, figsize=(16, 4))
+sns.kdeplot(
+    data=visit_combined, x="patient_id", hue="type", ax=axes[0], palette=map_color
+)
+day_order = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]
+visit_combined["day_visit"] = pd.Categorical(
+    visit_combined["day_visit"], categories=day_order, ordered=True
+)
+visit_combined = visit_combined.sort_values("day_visit")
+sns.countplot(
+    data=visit_combined, x="day_visit", hue="type", palette=map_color, ax=axes[1]
+)
+sns.countplot(data=visit_combined, x="exam", hue="type", palette=map_color, ax=axes[2])
+
+
+# ## Multivariate comparison
+#
+
+# ### Visit x Patient
+
+# +
+doctor_avatar_renamed = doctor_avatar.rename(columns={"age": "age_doctor"})
+visit_avatar_flat = visit_avatar.join(
+    doctor_avatar_renamed.set_index("doctor_id"), on="doctor_id"
+)
+visit_avatar_flat = visit_avatar_flat.join(
+    patient_avatar.set_index("patient_id"), on="patient_id"
+)
+
+
+doctor_renamed = doctor.rename(columns={"age": "age_doctor"})
+visit_flat = visit.join(doctor_renamed.set_index("doctor_id"), on="doctor_id")
+visit_flat = visit_flat.join(patient.set_index("patient_id"), on="patient_id")
+
+
+# -
+
+visit_flat["day_visit"] = pd.Categorical(
+    visit_flat["day_visit"], categories=day_order, ordered=True
+)
+visit_flat = visit_flat.sort_values("day_visit")
+visit_avatar_flat["day_visit"] = pd.Categorical(
+    visit_avatar_flat["day_visit"], categories=day_order, ordered=True
+)
+visit_avatar_flat = visit_avatar_flat.sort_values("day_visit")
+fig, axes = plt.subplots(1, 2, figsize=(12, 4))
+sns.countplot(
+    data=visit_flat,
+    x="day_visit",
+    hue="gender",
+    ax=axes[0],
+    palette=["dimgrey", "lightgrey"],
+)
+sns.countplot(
+    data=visit_avatar_flat,
+    x="day_visit",
+    hue="gender",
+    ax=axes[1],
+    palette=["#3BD6B0", "#9fe9d7"],
+)
+axes[0].set_title("Original")
+axes[1].set_title("Avatar")
+
+# ### Visit x Doctor
+
+visit_flat["day_visit"] = pd.Categorical(
+    visit_flat["day_visit"], categories=day_order, ordered=True
+)
+visit_flat = visit_flat.sort_values(["day_visit", "exam"])
+visit_avatar_flat["day_visit"] = pd.Categorical(
+    visit_avatar_flat["day_visit"], categories=day_order, ordered=True
+)
+visit_avatar_flat = visit_avatar_flat.sort_values(["day_visit", "exam"])
+fig, axes = plt.subplots(1, 2, figsize=(12, 4))
+sns.boxplot(
+    data=visit_flat,
+    x="age_doctor",
+    y="day_visit",
+    ax=axes[0],
+    palette=["dimgrey", "lightgrey"],
+)
+sns.boxplot(
+    data=visit_avatar_flat,
+    x="age_doctor",
+    y="day_visit",
+    ax=axes[1],
+    palette=["#3BD6B0", "#9fe9d7"],
+)
+axes[0].set_title("Original")
+axes[1].set_title("Avatar")
```

Comment:
It might be easier to read if the x-axis is the same size in the original and in the avatar.